miner/worker.go

```diff
@@ -598,8 +599,10 @@ func (w *worker) mainLoop() {
 			// In wemix, costly interrupt / resubmit is disabled
 			if wemixminer.IsPoW() {
 				w.commitWork(req.interrupt, req.noempty, req.timestamp)
-			} else {
+			} else if wemixminer.AmPartner() {
```

Comment:
I think it's better to modify line 539 rather than this line as following because we don't require inserting `newWork` if we would do like that.

```
if atomic.CompareAndSwapInt32(&busyMining, 0, 1)
=>
if atomic.CompareAndSwapInt32(&busyMining, 0, 1) && wemixminer.AmPartner()
```

Reply:
> ```
> if atomic.CompareAndSwapInt32(&busyMining, 0, 1) && wemixminer.AmPartner()
> ```

This makes `busyMining` fixed to `1` forever because `busyMining` is reset only in success case of CAS operation.
`busyMining` is not forcely reset to `0` if the value is `1`.
Checking the condition preceeded by CAS operation makes sense.

```
if atomic.CompareAndSwapInt32(&busyMining, 0, 1)
=>
if wemixminer.AmPartner() && atomic.CompareAndSwapInt32(&busyMining, 0, 1)
```

Reply:
nice. i involved it to work